pkg/apis/maistra/conversion/extension_providers_test.go

```diff
@@ -176,3 +177,38 @@ meshConfig:
 		},
 	}
 }
+
+func TestInvalidValueType(t *testing.T) {
+	helmValues := v1.HelmValues{}
+	if err := helmValues.UnmarshalYAML([]byte(`
+meshConfig:
+  extensionProviders:
+  - envoyExtAuthzHttp:
+      service: "test"
+      port: "80"
+`)); err != nil {
+		t.Fatalf("failed to parse helm values: %s", err)
+	}
+
+	specV1 := v1.ControlPlaneSpec{
+		Version: "v2.4",
+		Istio:   &helmValues,
+	}
+	var specV2 v2.ControlPlaneSpec
+	err := Convert_v1_ControlPlaneSpec_To_v2_ControlPlaneSpec(&specV1, &specV2, nil)
+	if err != nil {
+		t.Fatalf("failed to convert spec: %s", err)
```

Comment:
I don't understand why an error is unexpected... If `convertEnvoyExtAuthzHTTPValuesToConfig` returns error in case of failure during types casting, shouldn't an error be expected here?

Reply:
This [defer ](https://github.com/Maistra/istio-operator/blob/66ad6f6cac66c9fd277e239e2771b1dfce13ddcb/pkg/apis/maistra/conversion/conversion_v1_to_v2.go#L88-L121) function catches conversion errors and makes the function return nil on errors.

Reply:
I guess I could've just called the ` convertEnvoyExtAuthzHTTPValuesToConfig` function in the unit test to make the test simpler.

Reply:
Oh... I didn't notice it. Thanks.

Reply:
> I guess I could've just called the  convertEnvoyExtAuthzHTTPValuesToConfig function in the unit test to make the test simpler.

Then techPreview values would not be tested.